### ISSUE TITLE
Incorrect type in equals when using @EqualsAndHashCode

### ISSUE THREAD

**Issue by orange-buffalo** [rzwitserloot/lombok#1274]
Consider the following code:
```java
import lombok.EqualsAndHashCode;

public interface DynamicType {

    interface Unloaded extends DynamicType {

    }

    class Default implements DynamicType {

        @EqualsAndHashCode(callSuper = false)
        public static class Unloaded extends Default implements DynamicType.Unloaded {

        }
    }
}
```
Compilation fails with error:
```
DynamicType.java:[11,9] reference to Unloaded is ambiguous
[ERROR] both class DynamicType.Default.Unloaded in DynamicType.Default and interface DynamicType.Unloaded in DynamicType match
```
If we check the generated code
```java
...
public boolean equals(final java.lang.Object o) {
       if (o == this) return true;
       if (!(o instanceof DynamicType.Default.Unloaded)) return false;
       final Unloaded other = (Unloaded) o;
...
```
we can see that type name is not fully specified in cast operation. All other places of generated code use canonical class name.

**Comment by rzwitserloot** [comment]
Fixed in release 1.16.14. It's the 'you never encountered this bug' reference in the changelog :)

**Comment by orange-buffalo** [comment]
Cool, thank you!
